mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -57,7 +57,19 @@ sealed trait Vector extends Serializable {
     }
   }
 
-  override def hashCode(): Int = util.Arrays.hashCode(this.toArray)
+  override def hashCode(): Int = {
+    var result: Int = size + 31
+    this.foreachActive{case(index, value) =>
```

Comment:
You should put spaces around the `{` and in `if(...` below. I think you could note where this hash function comes from (the JDK right?) to explain it. But yes this is looking right.